<a href="https://colab.research.google.com/github/taceroc/DSPS_TAcero/blob/main/labs/Copy_of_MLNPS_Ztestpandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [25]:
import pandas as pd
import numpy as np
%pylab inline

Populating the interactive namespace from numpy and matplotlib


I extracted precipitations records for Willmington from 2000/01/01 through today. 
The goal is to extract from these data the daily precipitation starting on 2018/01/01 and assess if these data are different from a population with mean $\mu = 0.14$ standard deviation $\sigma= 0.38$. 

Set a p-value at $2\sigma$:  $p=0.05 \rightarrow 2\sigma$

$p = 0.007 \rightarrow 3\sigma$

In [16]:
rain = pd.read_csv("https://raw.githubusercontent.com/fedhere/DSPS/master/labs/1865331.csv")


The $Z$ statistics is : $Z = \frac{\bar{X} - \mu }{s}$ and $s = \sigma / \sqrt(N)$ (you should know what all those symbols mean)

1. Extract from the dataframe above the columns "DATE" and "PRCP" (precipitation)

2. Convert the dataframe DATE from ```string``` to ```datetime``` type
 using pd.datetime() 
    
3. Select the data rows with date later than 2018/01/01. 
    If your data type conversion worked well you should be able to set the condition as ```rain[<newcolumnname>]> pd.to_datetime("2018/01/01")```
    use the technique called "broadcasting" to select the rows: 
    
    ```rain[ rain[<newcolumnname>]> pd.to_datetime("2018/01/01")]``` which you can save as a new dataframe
    
4. Calculate the mean of the sample you created $\bar{X}$ and plug it in  the numerator of the $Z$ statistics: $\bar{X}- \mu$

5. Calculate the size of the sample and plug it in the formula. Now you have $Z$

6. Knowing the distribution of $Z\sim N(\mu=0, \sigma=1)$ and with a confidence $p=0.05$, did you find that the time period 2018/01/01---today has a statistically significant different amount of rain?  
    

In [28]:
rain = rain[["DATE", "PRCP"]]
rain["DATE"] = pd.to_datetime(rain["DATE"])

In [29]:
rain

DATE  PRCP
0    2010-01-01  0.28
1    2010-01-02  0.00
2    2010-01-03  0.00
3    2010-01-04  0.00
4    2010-01-05  0.00
...         ...   ...
3533 2019-09-04  0.00
3534 2019-09-05  0.00
3535 2019-09-06  0.00
3536 2019-09-07  0.02
3537 2019-09-08  0.00

[3538 rows x 2 columns]

In [30]:
rain.dtypes

DATE    datetime64[ns]
PRCP           float64
dtype: object

In [31]:
rain_new = rain[rain["DATE"]>pd.to_datetime("2018/01/01")]
rain_new

DATE  PRCP
2923 2018-01-02  0.00
2924 2018-01-03  0.00
2925 2018-01-04  0.07
2926 2018-01-05  0.20
2927 2018-01-06  0.00
...         ...   ...
3533 2019-09-04  0.00
3534 2019-09-05  0.00
3535 2019-09-06  0.00
3536 2019-09-07  0.02
3537 2019-09-08  0.00

[615 rows x 2 columns]

In [32]:
rain_18_mean = rain_new['PRCP'].mean()
rain_18_std = rain_new['PRCP'].std()
n = len(rain_new)
print("mean: {}, std: {}, len: {}".format(rain_18_mean,rain_18_std,n))

mean: 0.16250406504065049, std: 0.3650071253149441, len: 615


In [33]:
mu = 0.14
stds = 0.38
z = (rain_18_mean - mu) / (stds/np.sqrt(n)) 
z

1.4686385901934103

z = 1.4686385901934103 sigmas

H$_0$: These data are from a distribution of $\mu = 0.14$ and $\sigma = 0.38$

H$_1$: These data are from a distribution with NOT $\mu = 0.14$ and $\sigma = 0.38$

Prob = 1.47$\sigma$

p = 0.007 $\rightarrow 3\sigma$

Our threshold was 3$\sigma$, then Prob < p. I can reject the null hypothesis *** CHECK